tests/test_create.py

```diff
@@ -851,213 +847,51 @@ def test_rm_rf(clear_package_cache: None, tmp_env: TmpEnvFixture):
-    with make_temp_env("python=3.6", "flask=1.0.2", "bzip2=1.0.8") as prefix:
-        env_file = join(prefix, "env.yml")
-        touch(env_file)
-        with open(env_file, "w") as f:
-            f.write(
-                dals(
-                    """
-                    name: dummy
-                    channels:
-                      - defaults
-                    dependencies:
-                      - bzip2=1.0.8
-                      - flask>=1.0.1,<=1.0.4
-                    """
-                )
-            )
-        output, _, _ = run_command(Commands.COMPARE, prefix, env_file, "--json")
-        assert "Success" in output
-        rmtree(prefix, ignore_errors=True)
+def test_tarball_install(
+    test_recipes_channel: Path,
+    tmp_env: TmpEnvFixture,
+    conda_cli: CondaCLIFixture,
+):
+    with tmp_env(test_recipes_channel / "noarch" / "dependent-1.0-0.tar.bz2") as prefix:
+        assert package_is_installed(prefix, "dependent")
+        assert not package_is_installed(prefix, "dependency")
+        conda_cli("remove", f"--prefix={prefix}", "dependent", "--yes")
+        assert not package_is_installed(prefix, "dependent")
 
 
-def test_compare_fail():
-    with make_temp_env("python=3.6", "flask=1.0.2", "bzip2=1.0.8") as prefix:
-        env_file = join(prefix, "env.yml")
-        touch(env_file)
-        with open(env_file, "w") as f:
-            f.write(
-                dals(
-                    """
-                    name: dummy
-                    channels:
-                      - defaults
-                    dependencies:
-                      - yaml
-                      - flask=1.0.3
-                    """
-                )
-            )
-        output, _, _ = run_command(Commands.COMPARE, prefix, env_file, "--json")
-        assert "yaml not found" in output
-        assert (
-            "flask found but mismatch. Specification pkg: flask=1.0.3, Running pkg: flask==1.0.2=py36_1"
-            in output
-        )
-        rmtree(prefix, ignore_errors=True)
+def test_tarball_install_and_bad_metadata(
+    test_recipes_channel: Path, tmp_env: TmpEnvFixture, conda_cli: CondaCLIFixture
+):
+    with tmp_env("small-executable", "dependent", "another_dependent") as prefix:
+        assert package_is_installed(prefix, "another_dependent")
+        conda_cli("remove", f"--prefix={prefix}", "dependent", "--yes")
+        assert package_is_installed(prefix, "small-executable")
+        assert not package_is_installed(prefix, "dependent")
+        # make sure all dependencies of "dependent" were removed
+        assert not package_is_installed(prefix, "dependency")
+        assert not package_is_installed(prefix, "another_dependent")
 
+        tar_path = test_recipes_channel / "noarch/dependent-1.0-0.tar.bz2"
+        with pytest.raises(DryRunExit):
+            conda_cli("install", f"--prefix={prefix}", tar_path, "--dry-run")
+            assert not package_is_installed(prefix, "dependent")
```

Comment:
Since `conda_cli` will exit with a `DryRunExit` this assertion will never be run:

```suggestion
```